src/_macosx.m

```diff
@@ -2066,29 +2066,31 @@ - (BOOL)acceptsFirstResponder
     return YES;
 }
 
-/* This is all wrong. Address of pointer is being passed instead of pointer, keynames don't
-   match up with what the front-end and does the front-end even handle modifier keys by themselves?
-
+// flagsChanged gets called on single modifier keypresses
+// The modifier + second key gets handled in convertKeyEvent in KeyUp/KeyDown
```

Comment:
does it also work for e.g. "ctrl+shift"? (dunno, just asking)

Reply:
a ctrl, shift, f for example produces this:
```python
you pressed cmd None None
you pressed shift None None
you pressed cmd+F None None
you pressed cmd None None
```

I think the extra command at the end is there due to the shift also being let up and changing state while command is still pressed...

Reply:
On other backends, the second press is "cmd+shift", not just shift.  (Well, ctrl+shift on non-osx.)

Reply:
OK, so this was more complicated than I had hoped... The flagsChanged gets called on any press OR release, which is why I was printing out "pressed" when it was really a "release".

I think this means we have to keep track of the state of the keys ourselves and what is going up/down. I think this now matches what other backends do, including press/release properly.